### PR TITLE
[Sparse] Support spspdiv

### LINKED ISSUE #5369: [Sparse] Support SparseMatrix element-wise division with the same sparsity.
## 🔨Work Item

**IMPORTANT:**
* This template is only for dev team to track project progress. For feature request or bug report, please use the corresponding issue templates.
* DO NOT create a new work item if the purpose is to fix an existing issue or feature request. We will directly use the issue in the project tracker.

Project tracker: https://github.com/orgs/dmlc/projects/2

### PR DESCRIPTION
**Pull request by czkkkkkk**
## Description
Resolve #5369 .

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by jermainewang — APPROVED** [review]

File: python/dgl/sparse/elementwise_op_sp.py [hunk 1]
```diff
@@ -141,8 +148,9 @@ def sp_mul(A: SparseMatrix, B: Union[SparseMatrix, Scalar]) -> SparseMatrix:
 def sp_div(A: SparseMatrix, B: Union[SparseMatrix, Scalar]) -> SparseMatrix:
     """Elementwise division
 
-    If :attr:`B` is a sparse matrix, both :attr:`A` and :attr:`B` must be
-    diagonal matrices.
+    If :attr:`B` is a sparse matrix, both :attr:`A` and :attr:`B` must have the
+    same sparsity. And the returned matrix has the same order of non-zero
+    entries as :attr:`A`.
```
Comment (frozenbugs) [inline python/dgl/sparse/elementwise_op_sp.py:153]: Is it better to have the same order as A or just have the sorted order?

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/sparse/elementwise_op_sp.py (hunk 1 above)
Comment (czkkkkkk) [reply python/dgl/sparse/elementwise_op_sp.py:153]: I think it is better to have the same order as A. If users divide two matrices with the same nnz order, it is weird to return a matrix with another order.

File: python/dgl/sparse/elementwise_op_sp.py [hunk 2]
```diff
@@ -169,15 +177,7 @@ def sp_div(A: SparseMatrix, B: Union[SparseMatrix, Scalar]) -> SparseMatrix:
     """
     if is_scalar(B):
         return val_like(A, A.val / B)
-    if A.is_diag() and B.is_diag():
-        assert A.shape == B.shape, (
-            f"The shape of diagonal matrix A {A.shape} and B {B.shape} must"
-            f"match for elementwise division."
-        )
-        return diag(A.val / B.val, A.shape)
```
Comment (frozenbugs) [inline python/dgl/sparse/elementwise_op_sp.py:177]: Also remove import diag in L6.

**Comment by czkkkkkk** [comment]
@dgl-bot

**Pull request merged**